cpp/ScanChain_Zp.C

```diff
@@ -649,6 +675,36 @@ int ScanChain(TChain *ch, double genEventSumw, TString year, TString process) {
       double puppimet_pt  = puppimet.first;
       double puppimet_phi = puppimet.second;
 
+      // Account for tunepRelPt if useTuneP
+      vector<LorentzVector> Muon_p4 = {};
+      vector<double> Muon_pt = {};
+      vector<double> Muon_tkRelIso = {};
+      TVector2 pfmet_temp;
+      pfmet_temp.SetMagPhi(pfmet_pt,pfmet_phi);
+      for ( unsigned int im=0; im < nt.nMuon(); im++ ) {
+	Muon_p4.push_back(LorentzVector(nt.Muon_pt().at(im),nt.Muon_eta().at(im),nt.Muon_phi().at(im),nt.Muon_mass().at(im)));
+	Muon_pt.push_back(nt.Muon_pt().at(im));
+	Muon_tkRelIso.push_back(nt.Muon_tkRelIso().at(im));
+	if ( nt.Muon_highPtId().at(im)>=2 ) {
+	  if ( useTuneP ) {
+	    TVector2 muon_temp;
+	    TVector2 muon_temp_tunep;
+	    muon_temp.SetMagPhi(Muon_pt[im],nt.Muon_phi().at(im));
+	    Muon_p4[im] = LorentzVector(Muon_pt[im],nt.Muon_eta().at(im),nt.Muon_phi().at(im),nt.Muon_mass().at(im));
```

Comment:
Isn't it better to alter the pT even in the lorentz vector to its tuneP counterpart, especially since we are using this for subsequent calculation, e.g. Mμμ in line 1023?

As it is, I think we are still using the PF one. Probably we would like to switch the order of this and the next line?

Reply:
Yes, thanks for catching it. That was the intention.

Reply:
Done.